### ISSUE TITLE
Segfault in OpenTelemetrySpanLog

### ISSUE THREAD

**Issue by tavplubix** [ClickHouse/ClickHouse#38221]
https://s3.amazonaws.com/clickhouse-test-reports/0/a60a55b83cfb7a7fb1a4a0bba2af19fb5595cb83/stress_test__undefined__actions_.html

https://pastila.nl/?004a4e1e/09b34dbc34738fc97635d5f7d9c84333

Related to #37837, #38032, #38157

**Comment by tavplubix** [comment]
Hm, yes, it's BC check

**Comment by qoega** [comment]
Anyway it looks very wrong. I'll add at least logging what was the key without value.
```
attribute_names = {<std::__1::vector<DB::Field, AllocatorWithMemoryTracking<DB::Field> >> = {__begin_ = 0x7f7aeed89300, __end_ = 0x7f7aeed89338, __end_cap_ = {<std::__1::__compressed_pair_elem<DB::Field*, 0, false>> = {__value_ = 0x7f7aeed89338}, <std::__1::__compressed_pair_elem<AllocatorWithMemoryTracking<DB::Field>, 1, true>> = {<AllocatorWithMemoryTracking<DB::Field>> = {<No data fields>}, <No data fields>}, <No data fields>}}, <No data fields>}
attribute_values = {<std::__1::vector<DB::Field, AllocatorWithMemoryTracking<DB::Field> >> = {__begin_ = 0x0, __end_ = 0x0, __end_cap_ = {<std::__1::__compressed_pair_elem<DB::Field*, 0, false>> = {__value_ = 0x0}, <std::__1::__compressed_pair_elem<AllocatorWithMemoryTracking<DB::Field>, 1, true>> = {<AllocatorWithMemoryTracking<DB::Field>> = {<No data fields>}, <No data fields>}, <No data fields>}}, <No data fields>}}, <No data fields>}
```

**Comment by qoega** [comment]
Will run tests for now #38222
One bad attribute was removed in #38157

**Comment by tavplubix** [comment]
> I think it is fixed in master. 

Where?

**Comment by tavplubix** [comment]
I mean that BC check fails in master and I don't remember any fixes for it. Yes, it fails with previous version, but it fails. No checks should fail in master.

**Comment by qoega** [comment]
> Where?
This one https://github.com/ClickHouse/ClickHouse/pull/38157

**Comment by tavplubix** [comment]
This failure happened after #38157 was merged, so tests in master are still broken

**Comment by tavplubix** [comment]
Please also check the script that runs Stress Tests. AFAIK it downloads last released version, so it's not enough to backport the fix. Seems like we have to make a release to fix tests in master (and it's awful).

**Comment by tavplubix** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/38265/ed8341025b49b31a674a27e58b1837a50bb0ba59/stress_test__undefined__actions_.html

**Comment by tavplubix** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/0/e1823f4c0ab785fb2001fac62f5d29c5603224b1/stress_test__address__actions_.html

**Comment by tavplubix** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/0/4cda5491f639081428765181996cf1e118af5cfa/stress_test__undefined__actions_.html

**Comment by qoega** [comment]
1) We really need a release to use it in backward incompatibility check
2) 22.6 release branch is unstable and for 3rd day I can't press release
3) Revert will not help at all as release 22.6.1 will still be the same.

**Comment by tavplubix** [comment]
That's why I suggested to revert it **before** releasing 22.6...
Btw, there's another issue to fix: #38402

**Comment by tavplubix** [comment]
> https://s3.amazonaws.com/clickhouse-test-reports/0/be213c0be70bc53a5e04a0de87ef02f259a4640c/stress_test__thread__actions_.html

Hm, looks like in this case it crashed with 22.6.2 that contains patch that was supposed to fix the issue:
```
+ download_previous_release
INFO:root:Will download ReleaseInfo: 22.6.2.12-stable
```

**Comment by qoega** [comment]
I look at it again.
Probably there was something introduced with Map type support. Can't reproduce locally for now. And it is really strange why we saw it only in stress backward compatibility check.
https://github.com/ClickHouse/ClickHouse/commit/339133f32aba4c33e0ab90c3cb36c8f0ebdc72ed

```
(lldb) p this->attribute_names
(DB::Array) $3 = {
  DB::FieldVector = size=1 {
    [0] = {
      storage = (__lx = "thread_number\0\0\00T\xf1\U0000001c\0\0\0\rexecuteTask() CubeTransf")
      which = String
    }
  }
}
(lldb) p this->attribute_values
(DB::Array) $4 = {
  DB::FieldVector = size=0 {}
}
```

**Comment by tavplubix** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/0/c4155433349cdab8ff54584a982076030587e29e/stress_test__thread__actions_.html

**Comment by tavplubix** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/0/db949553193a8393caed01728d24c6fa2f7b4217/stress_test__thread__actions_.html

**Comment by tavplubix** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/38831/c9d1cbbc49d272fbfb769175585b2683e90e745e/stress_test__address__actions_.html

**Comment by qoega** [comment]
https://github.com/ClickHouse/ClickHouse/pull/38814

**Comment by tavplubix** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/0/2a6765ce8c8279af565ba54b4cf7b3e7f5a3f93f/stress_test__memory__actions_.html

**Comment by tavplubix** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/0/1bd97fa6fa7d3b1adafeec42f4ae2f6e50d43c41/stress_test__undefined__actions_.html
